### PR TITLE
Flag unused Refaster template parameters

### PR DESCRIPTION
**Pull request by Stephan202**

### CODE REVIEW

File: core/src/main/java/com/google/errorprone/bugpatterns/UnusedVariable.java [hunk 1]
```diff
@@ -149,6 +149,8 @@ public final class UnusedVariable extends BugChecker implements CompilationUnitT
   // TODO(ghm): Find a sensible place to dedupe this with UnnecessarilyVisible.
```
Comment (Stephan202) [inline core/src/main/java/com/google/errorprone/bugpatterns/UnusedVariable.java:149]: If acceptable I don't mind also updating [this list](https://github.com/google/error-prone/blob/b9e01a8d4f2a7a822eaf6b8f95f72f2f0ca72912/core/src/main/java/com/google/errorprone/bugpatterns/UnnecessarilyVisible.java#L59-L64). Alternatively we can upstream our [`RefasterRuleModifiers`](https://github.com/PicnicSupermarket/error-prone-support/blob/87f79dae60bd1be34f71ebeeba7d783670f96e70/error-prone-contrib/src/main/java/tech/picnic/errorprone/bugpatterns/RefasterRuleModifiers.java) check, which normalizes modifier usage by Refaster rule definitions more generally.

File: core/src/main/java/com/google/errorprone/bugpatterns/UnusedVariable.java [hunk 2]
```diff
@@ -149,6 +149,8 @@ public final class UnusedVariable extends BugChecker implements CompilationUnitT
...
   private static final ImmutableSet<String> ANNOTATIONS_INDICATING_PARAMETERS_SHOULD_BE_CHECKED =
       ImmutableSet.of(
+          "com.google.errorprone.refaster.annotation.AfterTemplate",
+          "com.google.errorprone.refaster.annotation.BeforeTemplate",
```
Comment (Stephan202) [inline core/src/main/java/com/google/errorprone/bugpatterns/UnusedVariable.java:153]: Of this list, currently there's only a test for `javax.inject.Inject`, so I didn't add any test cases for these new annotations, either. Let me know if I should. (I did test these changes against a collection of real Refaster templates.)

File: core/src/main/java/com/google/errorprone/bugpatterns/UnusedVariable.java (hunk 1 above)
Comment (cushon) [reply core/src/main/java/com/google/errorprone/bugpatterns/UnusedVariable.java:149]: > If acceptable I don't mind also updating [this list](https://github.com/google/error-prone/blob/b9e01a8d4f2a7a822eaf6b8f95f72f2f0ca72912/core/src/main/java/com/google/errorprone/bugpatterns/UnnecessarilyVisible.java#L59-L64).

Thanks, updating the list until we get around to deduping them SGTM

> Alternatively we can upstream our [`RefasterRuleModifiers`](https://github.com/PicnicSupermarket/error-prone-support/blob/87f79dae60bd1be34f71ebeeba7d783670f96e70/error-prone-contrib/src/main/java/tech/picnic/errorprone/bugpatterns/RefasterRuleModifiers.java) check, which normalizes modifier usage by Refaster rule definitions more generally.

That also sounds good if you have time/interest, but it also doesn't need to block this PR

**Review by cushon — CHANGES_REQUESTED** [review]
LG pending the update to UnnecessarilyVisible

File: core/src/main/java/com/google/errorprone/bugpatterns/UnusedVariable.java (hunk 1 above)
Comment (Stephan202) [reply core/src/main/java/com/google/errorprone/bugpatterns/UnusedVariable.java:149]: > Thanks, updating the list until we get around to deduping them SGTM

Top; will do.

> That also sounds good if you have time/interest, but it also doesn't need to block this PR

I'll make a note to consider this later then :heavy_check_mark:. (When we contributed #3813 and #2232 that caused some issues for downstream users due to a name collision (PicnicSupermarket/error-prone-support#622 and PicnicSupermarket/error-prone-support#686), so maybe this time I'll try to find an alternative name; hehe.)

**Review by Stephan202 — COMMENTED** [review]
Rebased and added a commit to also update `UnnecessarilyVisible`.

**Review by cushon — APPROVED** [review]

**Pull request closed without merging**